src/gamestate.ts

```diff
@@ -1,7 +1,59 @@
+import { areCoordinatesValid } from "./user";
+import { Coordinates } from "./utils";
+
 export interface GameState {
   board: Array<Array<Cell>>;
 }
 
 export interface Cell {
   value: "empty" | "black";
 }
+
+export function generateNewBoard(): GameState {
+  return {
+    board: [
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+    ],
+  };
+}
```

Comment:
Suggestion (untested):

```suggestion
export function generateNewBoard(n: number): GameState {
  return { board: Array(n).fill(Array(n).fill({ value: "empty" })) };
}
```